PWGMM/Tasks/analyse-mft-tracks.cxx

```diff
@@ -0,0 +1,114 @@
+      {"NtrkZvtx", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}},                                                         //
+      {"NtrkEta", "; N_{trk}; #eta; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {35, -4.5, -1.}}}},                                                                //
+      {"Multiplicity", "alibi_nightlies/O2DPG_pp_minbias_testbeam.sh/15-11-2021-18:00 - tf13; collisionID; N_{trk}^{MFT}", {HistType::kTH1F, {{101, 0., 100.}}}, true} //
+    }                                                                                                                                                                  //
+  };
+
+  void process(o2::aod::Collision const& collision, o2::aod::MFTTracks const& tracks)
+  {
+
+    auto z = collision.posZ();
+    registry.fill(HIST("NtrkZvtx"), tracks.size(), z);
+
+    for (auto& track : tracks) {
+      registry.fill(HIST("TracksPhiEta_in_coll"), track.phi(), track.eta());
+      registry.fill(HIST("TracksEtaZvtx"), track.eta(), z);
+      registry.fill(HIST("Multiplicity"), icoll);
+      registry.fill(HIST("NtrkEta"), tracks.size(), track.eta());
+    }
+    icoll++;
+  }
+  //end of process
+
+  //aod::McCollisions
+};
+//end of MyTask
+
+struct analyseGenTracks {
+
+  HistogramRegistry registryGen{
+    "registryGen",
+    {
+      {"TracksPhiEtaGen", "; #varphi; #eta; tracks", {HistType::kTH2F, {{600, 0, 2 * M_PI}, {35, -4.5, -1.}}}},   //
+      {"TracksEtaZvtxGen", "; #eta; Z_{vtx}; tracks", {HistType::kTH2F, {{35, -4.5, -1.}, {201, -20.1, 20.1}}}},  //
+      {"NtrkZvtxGen", "; N_{trk}; Z_{vtx}; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {201, -20.1, 20.1}}}}, //
+      {"NtrkEtaGen", "; N_{trk}; #eta; events", {HistType::kTH2F, {{301, -0.5, 300.5}, {35, -4.5, -1.}}}},        //
+    }                                                                                                             //
+  };
+
+  void process(aod::McCollisions::iterator const& mcCollision, Particles const& particles)
+  {
+    int nChargedPrimaryParticles = 0;
+    auto z = mcCollision.posZ();
+
+    for (auto& particle : particles) {
+      auto p = TDatabasePDG::Instance()->GetParticle(particle.pdgCode());
+      int charge = 0;
+      if (p == nullptr) {
+        // unknown particles will be skipped
+        if (particle.pdgCode() > 1000000000) {
+          //          auto x = (std::trunc(particle.pdgCode() / 10000) - 100000);
+          //          charge = x - std::trunc(x / 1000) * 1000;
+          LOGF(debug, "[{}] Nucleus with PDG code {}", particle.globalIndex(), particle.pdgCode() /*, charge*/); // (charge %d)
+        } else {
+          LOGF(debug, "[{}] Unknown particle with PDG code {}", particle.globalIndex(), particle.pdgCode());
+        }
+      } else {
+        charge = p->Charge();
+      }
+      if (charge != 0 && particle.isPhysicalPrimary()) {
```

Comment:
You probably want to select particles in MFT acceptance?

Reply:
For the moment no, since the eta region of the MFT is changing with the z_vtx